proposals/records/2-settings-unification.md

```diff
@@ -0,0 +1,115 @@
+`clap` and manual solutions for parsing env vars.
+
+The goal here is to have an easy way to document all settings on the system
+and have those settings work equally correctly from env vars and charts values.
+
+### Kubernetes Integration:
+Kubernetes integration is done through helm chart values settings wiring. All the
+environment variables are wired through various chart file values that are mapped
+to environment variables. This can make it difficult and confusing to know
+exactly how to configure the component as the user need to know the chart
+config as well as the destination desired env var configs they influence.
+
+### Standalone Integration:
+Some settings can be used through command line arguments for the standalone server
+but not all. Moreover, some of the settings have slightly different implementations
+for the allowed values.
+
+### Naming Conventions:
+No explicit naming conventions used across use cases and there are some inconsistencies.
+
+For example, in some places, namespacing is used but in most places, it is not used.
+
+
+## Proposed Solution
+
+The proposal is to implement naming conventions and utilize consistency across
+all uses of settings with NucliaDB.
+
+### Naming conventions
+
+
+#### Namespacing
+
+When a component needs to be configured, a namespace should be used that describes the usage
+and dependency.
+
+Good example: audit settings
+Example 1 that needs fixing: chitchat `LISTEN_PORT` -> `CHITCHAT_LISTEN_PORT` and `SEEDS` -> `CHITCHAT_SEEDS`
+Example 2 that needs fixing: `RUNNING_PORT`-> `HTTP_PORT` and `RUNNING_HOST` -> `HTTP_PORT`
+Example 3 that needs fixing: `NUCLIADB_INGEST` -> `INGEST_HTTP_ADDR`
+
+Settings where it makes sense to not use namespacing are ones where it makes sense
+to reuse them between different components. For example, `HTTP_HOST`
+for an http service or `GRPC_HOST` for a GRPC service.
+
+
+### Python integration
+
+- Continue using pydantic
+- All Python component settings should be defined in a `settings.py` file at the root of the module so they are easy to discover
+  (One example of this being broken is with featureflagging.py)
+- All Rust component settings should likewise be defined in a `settings.rs` at the root of the module
+
+
+### Reuse
+
+Make sure to have consistency between like settings. For example, HTTP and GRPC services
+should utilize the same env vars.
+
+### Kubernetes/Helm Integration
+
+- Do not map chart config settings -> env variables in a config map
+- Map env variables directly from helm chart configs so they are completely consistent
+
+
+Example values.yaml:
+
+```yaml
+HTTP_HOST: "0.0.0.0"
+HTTP_PORT: 8080
+METRICS_HTTP_HOST: "0.0.0.0"
+METRICS_HTTP_PORT: 8080
+SENTRY_URL:
+SEARCH_TIMEOUT: ""
+JAEGER_ENABLED: true
+CHITCHAT_BINDING_PORT: 31337
+CHITCHAT_BINDING_HOST: "0.0.0.0"
+...
+```
+
+The rest of the helm config values would be focused around configuring kubernetes deploys/settings/etc,
```

Comment:
So then, we'll have something like an `environment` in the values with all env variables in it?

Reply:
Correct. I'll update this proposal with some examples.